manifest.go

```diff
@@ -0,0 +1,655 @@
+	Metadata() []byte
+	// PartitionList returns a list of field summaries for each partition
+	// field in the spec. Each field in the list corresponds to a field in
+	// the manifest file's partition spec.
+	PartitionList() []fieldSummary
+
+	// HasAddedFiles returns true if AddedDataFiles > 0 or if it was null.
+	HasAddedFiles() bool
+	// HasExistingFiles returns true if ExistingDataFiles > 0 or if it was null.
+	HasExistingFiles() bool
+	// FetchEntries reads the manifest list file to fetch the list of
+	// manifest entries using the provided file system IO interface.
+	// If discardDeleted is true, entries for files containing deleted rows
+	// will be skipped.
+	FetchEntries(fs iceio.IO, discardDeleted bool) ([]ManifestEntry, error)
+}
+
+// ReadManifestList reads in an avro manifest list file and returns a slice
+// of manifest files or an error if one is encountered.
+func ReadManifestList(in io.Reader) ([]ManifestFile, error) {
+	dec, err := ocf.NewDecoder(in)
+	if err != nil {
+		return nil, err
+	}
+
+	out := make([]ManifestFile, 0)
+
+	for dec.HasNext() {
+		var file ManifestFile
+		if string(dec.Metadata()["format-version"]) == "2" {
+			file = &manifestFileV2{}
+		} else {
+			file = &manifestFileV1{}
+		}
+
+		if err := dec.Decode(file); err != nil {
+			return nil, err
+		}
+		out = append(out, file)
+	}
+
+	return out, dec.Error()
+}
+
+// ManifestEntryStatus defines constants for the entry status of
+// existing, added or deleted.
+type ManifestEntryStatus int8
+
+const (
+	EntryStatusEXISTING ManifestEntryStatus = 0
+	EntryStatusADDED    ManifestEntryStatus = 1
+	EntryStatusDELETED  ManifestEntryStatus = 2
+)
+
+// ManifestEntryContent defines constants for the type of file contents
+// in the file entries. Data, Position based deletes and equality based
+// deletes.
+type ManifestEntryContent int8
+
+const (
+	EntryContentData       ManifestEntryContent = 0
+	EntryContentPosDeletes ManifestEntryContent = 1
+	EntryContentEqDeletes  ManifestEntryContent = 2
+)
+
+// FileFormat defines constants for the format of data files.
+type FileFormat string
+
+const (
+	AvroFile    FileFormat = "AVRO"
+	OrcFile     FileFormat = "ORC"
+	ParquetFile FileFormat = "PARQUET"
+)
+
+type colMap[K, V any] struct {
+	Key   K `avro:"key"`
+	Value V `avro:"value"`
+}
+
+func avroColMapToMap[K comparable, V any](c *[]colMap[K, V]) map[K]V {
+	if c == nil {
+		return nil
+	}
+
+	out := make(map[K]V)
+	for _, data := range *c {
+		out[data.Key] = data.Value
+	}
+	return out
+}
+
+type dataFile struct {
+	Content          ManifestEntryContent   `avro:"content"`
+	Path             string                 `avro:"file_path"`
+	Format           FileFormat             `avro:"file_format"`
+	PartitionData    map[string]any         `avro:"partition"`
+	RecordCount      int64                  `avro:"record_count"`
+	FileSize         int64                  `avro:"file_size_in_bytes"`
+	BlockSizeInBytes int64                  `avro:"block_size_in_bytes"`
```

Comment:
I think this shouldn't be written for V2

Reply:
Since the fields don't exist in the defined schema for V2 they shouldn't get written when writing the file since we write according to the schema